src/lib.rs

```diff
@@ -1539,7 +1544,10 @@ impl GooseAttack {
             users.push(user);
             self.stats.users += 1;
             debug!("sleeping {:?} milliseconds...", sleep_duration);
-            tokio::time::delay_for(sleep_duration).await;
+            tokio::time::delay_for(sleep_duration - user_thread_started.elapsed()).await;
```

Comment:
I think this belongs in a small async helper function - because it should also not sleep a negative time.

Also: Could consider printing the drift corrected sleep_duration.

Reply:
I could do it at `debug!()` -- when writing this I started with printing it and because of the granularity of `.elapsed()` there's always a very small drift (microseconds), so it's a lot of "noise" in the end. Moving it into a helper makes sense.

Reply:
BTW: My understanding is the duration between two instances is already guaranteed to never be negative: https://doc.rust-lang.org/std/time/struct.Instant.html

Reply:
Ah, I see what you mean: if drift gets really big. Okay, handled in `util::sleep_minus_drift()`